graphs/src/anemoi/graphs/nodes/attributes/area_weights.py

```diff
@@ -214,7 +214,7 @@ def __init__(
     def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
         assert self.mask_node_attr_name in nodes, f"Node attribute '{self.mask_node_attr_name}' not found in nodes."
         attr_values = super().get_raw_values(nodes, **kwargs)
```

Comment:
```suggestion
        attr_values = super().get_raw_values(nodes, **kwargs).to(self.device)
```

I'm not sure the proposed change fixes the issue.

When I run the failing acceptance test myself with `pytest training/tests/integration -k test_training_cycle_stretched --longtest` I still get a failing test.

If I look at `mask.device` and `attr_values.device` (without your change), I get that `mask` is on the `cuda`, and `attr_values` is on `cpu` (`self.device` is `cuda`) – so I _think_ the correct change is to leave mask where it is (as the `.to` has no effect), and instead move `attr_values`.